setup.py

```diff
@@ -53,6 +53,8 @@ def version(path):
     "nbformat",
     "nbconvert<7.0.0",
     "requests",
+    "fsspec",
+    "universal_pathlib>=0.0.18",
```

Comment:
pandas need to be moved in the minimal set now that you are using pandas.io.common in a critical place.